applications/nrf5340_audio/src/bluetooth/bt_renderer/bt_rend.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "bt_rend.h"
+
+#include <zephyr/zbus/zbus.h>
+
+#include "bt_volume_control.h"
+#include "nrf5340_audio_common.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_renderer, CONFIG_BT_RENDERER_LOG_LEVEL);
+
+ZBUS_CHAN_DEFINE(volume_chan, struct volume_msg, NULL, NULL, ZBUS_OBSERVERS_EMPTY,
+		 ZBUS_MSG_INIT(0));
+
+int bt_rend_volume_up(void)
+{
+	int ret;
+	struct volume_msg msg;
+
+	if (IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR) || IS_ENABLED(CONFIG_BT_VCP_VOL_REND)) {
+		ret = bt_vol_ctrl_up();
+		return ret;
+	}
+
+	msg.event = VOLUME_UP;
+
+	ret = zbus_chan_pub(&volume_chan, &msg, K_NO_WAIT);
+	return ret;
+}
+
+int bt_rend_volume_down(void)
+{
+	int ret;
+	struct volume_msg msg;
+
+	if (IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR) || IS_ENABLED(CONFIG_BT_VCP_VOL_REND)) {
+		ret = bt_vol_ctrl_down();
+		return ret;
+	}
+
+	msg.event = VOLUME_DOWN;
+
+	ret = zbus_chan_pub(&volume_chan, &msg, K_NO_WAIT);
+	return ret;
+}
+
+int bt_rend_volume_set(uint8_t volume, bool from_vcp)
+{
+	int ret;
+	struct volume_msg msg;
+
+	if ((IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR) || IS_ENABLED(CONFIG_BT_VCP_VOL_REND)) &&
+	    !from_vcp) {
+		ret = bt_vol_ctrl_set(volume);
+		return ret;
+	}
+
+	msg.event = VOLUME_SET;
+	msg.value = volume;
+
+	ret = zbus_chan_pub(&volume_chan, &msg, K_NO_WAIT);
+	return ret;
+}
+
+int bt_rend_mute(bool from_vcp)
+{
+	int ret;
+	struct volume_msg msg;
+
+	if ((IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR) || IS_ENABLED(CONFIG_BT_VCP_VOL_REND)) &&
+	    !from_vcp) {
+		ret = bt_vol_ctrl_mute();
+		return ret;
+	}
+
+	msg.event = VOLUME_MUTE;
+
+	ret = zbus_chan_pub(&volume_chan, &msg, K_NO_WAIT);
+	return ret;
+}
+
+int bt_rend_unmute(void)
+{
+	int ret;
+	struct volume_msg msg;
+
+	if (IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR) || IS_ENABLED(CONFIG_BT_VCP_VOL_REND)) {
+		ret = bt_vol_ctrl_unmute();
+		return ret;
+	}
+
+	msg.event = VOLUME_UNMUTE;
+
+	ret = zbus_chan_pub(&volume_chan, &msg, K_NO_WAIT);
+	return ret;
+}
+
+int bt_rend_discover(struct bt_conn *conn)
+{
+	int ret;
+
+	if (IS_ENABLED(CONFIG_BT_VCP_VOL_CTLR)) {
+		ret = bt_vol_ctrl_vcs_discover(conn);
+		if (ret) {
+			LOG_WRN("Failed to discover VCS: %d", ret);
+			return ret;
+		}
+	}
```

Comment:
We can have a LOG_DBG. Since this is called from security_changed it will be called for both headset and gateway, even though gateway is the only volume controller. For other services (to come) it might be different.